api/src/main/java/org/motechproject/nms/api/web/KilkariController.java

```diff
@@ -101,4 +104,73 @@ public void saveInboxCallDetails(@RequestBody InboxCallDetailsRequest request) {
         //todo: the work...
     }
 
+    /**
+     * 4.2.3
+     * Create Subscription
+     *
+     * IVR shall invoke this API to create a new Kilkari subscription
+     *
+     */
+    @RequestMapping(value = "/subscription",
+            method = RequestMethod.POST,
+            headers = { "Content-type=application/json" })
+    @ResponseStatus(HttpStatus.OK)
+    public void createSubscription(@RequestBody SubscriptionRequest subscriptionRequest)
+        throws NotFoundException {
```

Comment:
He's already fixed that in his branch which he hasn't pushed yet, so I figured no need to also do it here.

-----Original Message-----
From: "Rob LaRubbio" <notifications@github.com>
Sent: ‎4/‎22/‎2015 4:39 PM
To: "koshalt/mim" <mim@noreply.github.com>
Cc: "Lauren Lavoie" <llavoie@grameenfoundation.org>
Subject: Re: [mim] Kilkari CreateSubscription and DeactivateSubscription (#73)

In api/src/main/java/org/motechproject/nms/api/web/KilkariController.java:
> @@ -101,4 +104,73 @@ public void saveInboxCallDetails(@RequestBody InboxCallDetailsRequest request) {
>          //todo: the work...
>      }
>  
> +    /**
> +     * 4.2.3
> +     * Create Subscription
> +     *
> +     * IVR shall invoke this API to create a new Kilkari subscription
> +     *
> +     */
> +    @RequestMapping(value = "/subscription",
> +            method = RequestMethod.POST,
> +            headers = { "Content-type=application/json" })
> +    @ResponseStatus(HttpStatus.OK)
> +    public void createSubscription(@RequestBody SubscriptionRequest subscriptionRequest)
> +        throws NotFoundException {
Remember Frank said the throws can be removed if NotFoundException extends IllegalStateException
—
Reply to this email directly or view it on GitHub.